File: app/src/main/java/com/joshrap/liteweight/repositories/workouts/requests/RestartWorkoutRequest.java
package com.joshrap.liteweight.repositories.workouts.requests;

import com.joshrap.liteweight.models.workout.Routine;
import com.joshrap.liteweight.repositories.BodyRequest;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class RestartWorkoutRequest extends BodyRequest {

    private SetRoutineRequest routine;

    public RestartWorkoutRequest(Routine routine) {
        this.routine = new SetRoutineRequest(routine);
    }
}

File: api/LiteWeightApi/Api/Workouts/Requests/RestartWorkoutRequest.cs
using System.ComponentModel.DataAnnotations;

namespace LiteWeightAPI.Api.Workouts.Requests;

public class RestartWorkoutRequest
{
	/// <summary>
	/// Workout to restart.
	/// </summary>
	[Required]
	public SetRoutineRequest Routine { get; set; } = null!;
}